server/aesdsocket.c

```diff
@@ -141,62 +150,114 @@ struct thread_data
 */
 static int process_recv_pkt(char **pkt, int clientFD, pthread_mutex_t *mutex, long len)
 {
-    //pthread_mutex_unlock(mutex);
     if(pkt == NULL)
     {
         syslog(LOG_ERR, "Passed Packet returned NULL pointer");
         return -1;
     }
-    char *readback  = (char *)malloc(total_len);
-    long i = 0;
 
+    printf("\n\t\tOpening device...");
     fd = open(OUTPUT_FILE, O_RDWR|O_CREAT|O_APPEND, S_IRWXU|S_IRWXG|S_IRWXO);
 
-    printf("\n%ld bytes to be written: %s,\n\n", len, *pkt);
-    printf("\nlocking\n");
+    printf("\n\t\t%ld bytes received: %s", len, *pkt);
+
+/* ioctl seekto struct */
+    struct aesd_seekto seekto;
+    char wr_cmd;
+    char wr_cmd_ofs;
+    bool ioctl_flag = false;
+
+    char find_seek[IOCTL_STRING_SIZE];
+    memcpy(find_seek, *pkt, IOCTL_STRING_SIZE - 1);
+    find_seek[IOCTL_STRING_SIZE] = '\0';
+
+    if(strcmp(find_seek, IOCTL_STRING) == 0)
+    {
+        printf("\n\t\tIOCTL Command Received. Parsing values...");
+        /* EXTRA BYTE??? */
+        total_len -= (len + NULL_TERM_BYTE+NULL_TERM_BYTE);
+        memcpy(find_seek, *pkt, len);
+
+        wr_cmd = find_seek[IOCTL_STRING_SIZE - 1];
+        wr_cmd_ofs = find_seek[IOCTL_STRING_SIZE + 1];
+        //printf("\n\t\twr_cmd: %d", wr_cmd);
+        //printf("\n\t\twr_cmd_ofs: %d", wr_cmd_ofs);
+
+        /* get X and Y, bytes 19 and 21 */
+        printf("\n\t\tconverting to ints...");
+        seekto.write_cmd = wr_cmd - CHAR_TO_INT;
+        seekto.write_cmd_offset = wr_cmd_ofs - CHAR_TO_INT;
```

Comment:
@mash9756 what does this assume about the number of digits in `write_cmd_offset` and `write_cmd`?  Is this a safe assumption?